apps/editing-toolkit/editing-toolkit-plugin/wpcom-block-editor-welcome-tour/src/tour-launch.js

```diff
@@ -78,24 +89,31 @@ function WelcomeTourFrame() {
 		<div className="wpcom-editor-welcome-tour-frame">
 			{ ! isMinimized ? (
 				<WelcomeTourCard
-					cardContent={ cardContent[ currentCard ] }
-					cardIndex={ currentCard }
-					key={ currentCard }
+					cardContent={ cardContent[ currentCardIndex ] }
+					cardIndex={ currentCardIndex }
+					justMaximized={ justMaximized }
+					key={ currentCardIndex }
 					lastCardIndex={ cardContent.length - 1 }
 					onDismiss={ dismissWpcomNuxTour }
 					onMinimize={ setIsMinimized }
-					setCurrentCard={ setCurrentCard }
+					setJustMaximized={ setJustMaximized }
+					setCurrentCardIndex={ setCurrentCardIndex }
 				/>
 			) : (
-				<WelcomeTourMinimized onMaximize={ setIsMinimized } />
+				<WelcomeTourMinimized onMaximize={ setIsMinimized } setJustMaximized={ setJustMaximized } />
 			) }
 		</div>
 	);
 }
 
-function WelcomeTourMinimized( { onMaximize } ) {
+function WelcomeTourMinimized( { onMaximize, setJustMaximized } ) {
+	const handleOnMaximize = () => {
```

Comment:
Looks like you also use the `onAction`/`handleAction` naming convention! I like it too :D Keep it up and we will eventually get it through the whole codebase 😅